scipy/stats/_multivariate.py

```diff
@@ -309,7 +313,11 @@ class multivariate_normal_gen(multi_rv_generic):
     -----
     %(_mvn_doc_callparams_note)s
 
-    The covariance matrix `cov` must be a symmetric positive semidefinite
+    The covariance matrix `cov` may be an instance of a subclass of
+    `Covariance`, e.g. `scipy.stats.CovViaPrecision`. If so, `allow_singular`
```

Comment:
I don't think the user knows what a "Covariance" object is. Probably better to just say `CovViaPrecision`.

Reply:
> I don't think the user knows what a "Covariance" object is

How do we fix that? In Java, there would be a `Covariance` interface, and any concrete implementation of that interface would be acceptable. That's what I'm going for here.

Reply:
We can make the Covariance class public. It will also allow users to extend it. Maybe a bit soon now but when we have more covariance objects, we can do that.